py/picca/io.py

```diff
@@ -1033,7 +1047,10 @@ def read_from_desi(in_dir, catalog, pk1d=None):
                 flux = spec['FL'][w_t].copy()
 
                 if not pk1d is None:
-                    reso_sum = spec['RESO'][w_t].copy()
+                    if not reso_from_truth:
+                        reso_sum = spec['RESO'][w_t].copy()
```

Comment:
what is w_t and where is it defined?

Reply:
`w_t = np.where(targetid_spec == entry[id_name])[0]` from ~15 lines further up, it's the index of the currently used spectrum (RESO is a (nspec,ndiag,npix) array). If `reso_from_truth` was set RESO is only an (ndiag,npix) array and the selection is not needed

Reply:
If reso does not exist in any form at this point this will fail.

Reply:
ok